src/epoll.rs

```diff
@@ -30,48 +30,47 @@ impl Poller {
         const CLOEXEC: libc::c_int = libc::EPOLL_CLOEXEC;
 
         // Create an epoll instance.
-        let epoll_fd = unsafe {
-            // Check if the `epoll_create1` symbol is available on this platform.
-            let ptr = libc::dlsym(
-                libc::RTLD_DEFAULT,
-                "epoll_create1\0".as_ptr() as *const libc::c_char,
-            );
-
-            if ptr.is_null() {
-                // If not, use `epoll_create` and manually set `CLOEXEC`.
-                let fd = match libc::epoll_create(1024) {
-                    -1 => return Err(io::Error::last_os_error()),
-                    fd => fd,
-                };
-                let flags = libc::fcntl(fd, libc::F_GETFD);
-                libc::fcntl(fd, libc::F_SETFD, flags | libc::FD_CLOEXEC);
-                fd
-            } else {
-                // Use `epoll_create1` with `CLOEXEC`.
-                let epoll_create1 = std::mem::transmute::<
-                    *mut libc::c_void,
-                    unsafe extern "C" fn(libc::c_int) -> libc::c_int,
-                >(ptr);
-                match epoll_create1(CLOEXEC) {
-                    -1 => return Err(io::Error::last_os_error()),
-                    fd => fd,
+        //
+        // Use `epoll_create1` with `CLOEXEC`.
+        let epoll_fd = syscall!(syscall(libc::SYS_epoll_create1, CLOEXEC))
```

Comment:
Is there any point in defining `const CLOEXEC` above if it's only used here? We could just put `libc::EPOLL_CLOEXEC` here, right?

Reply:
correct